tests/integration/test_schema_registry_auth.py

```diff
@@ -0,0 +1,215 @@
+    subject = new_random_name("any-")
+
+    res = await registry_async_client_auth.post(
+        f"compatibility/subjects/{quote(subject)}/versions/1", json={"schema": schema_avro_json}
+    )
+    assert res.status_code == 401
+
+    res = await registry_async_client_auth.get(f"config/{quote(subject)}")
+    assert res.status_code == 401
+
+    res = await registry_async_client_auth.put(f"config/{quote(subject)}", json={"compatibility": "NONE"})
+    assert res.status_code == 401
+
+    res = await registry_async_client_auth.get("config")
+    assert res.status_code == 401
+
+    res = await registry_async_client_auth.put("config", json={"compatibility": "NONE"})
+    assert res.status_code == 401
+
+    res = await registry_async_client_auth.get("schemas/ids/1/versions")
+    assert res.status_code == 401
+
+    res = await registry_async_client_auth.get("schemas/types")
+    assert res.status_code == 200
+
+    res = await registry_async_client_auth.get("schemas/types", auth=admin)
+    assert res.status_code == 200
+
+    res = await registry_async_client_auth.post(f"subjects/{quote(subject)}/versions", json={"schema": schema_avro_json})
+    assert res.status_code == 401
+
+    res = await registry_async_client_auth.delete(f"subjects/{quote(subject)}/versions/1")
+    assert res.status_code == 401
+
+    res = await registry_async_client_auth.get(f"subjects/{quote(subject)}/versions/1/schema")
+    assert res.status_code == 401
+
+    res = await registry_async_client_auth.delete(f"subjects/{quote(subject)}")
+    assert res.status_code == 401
+
+
+async def test_sr_list_subjects(registry_async_client_auth: Client) -> None:
+    cavesubject = new_random_name("cave-")
+    carpetsubject = new_random_name("carpet-")
+
+    res = await registry_async_client_auth.post(
+        f"subjects/{quote(cavesubject)}/versions", json={"schema": schema_avro_json}, auth=aladdin
+    )
+    assert res.status_code == 200
+    sc_id = res.json()["id"]
+    assert sc_id >= 0
+
+    res = await registry_async_client_auth.post(
+        f"subjects/{quote(carpetsubject)}/versions", json={"schema": schema_avro_json}, auth=admin
+    )
+    assert res.status_code == 200
+
+    res = await registry_async_client_auth.get("subjects", auth=admin)
+    assert res.status_code == 200
+    assert [cavesubject, carpetsubject] == res.json()
+
+    res = await registry_async_client_auth.get(f"subjects/{quote(carpetsubject)}/versions")
+    assert res.status_code == 401
+
+    res = await registry_async_client_auth.get(f"subjects/{quote(carpetsubject)}/versions", auth=admin)
+    assert res.status_code == 200
+    assert [sc_id] == res.json()
+
+    res = await registry_async_client_auth.get("subjects", auth=aladdin)
+    assert res.status_code == 200
+    assert [cavesubject] == res.json()
+
+    res = await registry_async_client_auth.get(f"subjects/{quote(carpetsubject)}/versions", auth=aladdin)
+    assert res.status_code == 403
```

Comment:
The spec (in my understanding) only refers to `GET /subjects` which lists all available subjects, not all `/subjects/something` prefixed endpoints. This exact endpoint does not leak any information, because access is checked first based on only requested path without reading the database.

Reply:
> This exact endpoint does not leak any information, because access is checked first based on only requested path without reading the database.

Ah, I see. So it means if a subject doesn't exist (and because of this we don't have an ACL record for it), it will also be 403, right?

Reply:
Yes, this is now 403 if no access regardless if a subject exists or not. Subject data is only checked if the user has access to it.